File: app/src/main/java/it/unical/mat/embasp/specializations/solver_planning_domains/SPDPlan.java
package it.unical.mat.embasp.specializations.solver_planning_domains;

import org.json.simple.DeserializationException;
import org.json.simple.JsonArray;
import org.json.simple.JsonObject;
import org.json.simple.Jsoner;

import it.unical.mat.embasp.languages.pddl.Action;
import it.unical.mat.embasp.languages.pddl.Plan;

public class SPDPlan extends Plan {

	public SPDPlan(final String plan, final String error) {
		super(plan, error);
	}

	@Override
	protected void parse() {
		if (errors != "" || output == "")
			return;
		JsonObject obj;
		try {
			obj = (JsonObject) Jsoner.deserialize(output);
			final String status = (String) obj.get("status");
			if (status.contains("ok")) {
				final JsonArray arrayPlan = (JsonArray) ((JsonObject) obj.get("result")).get("plan");
				for (int i = 0; i < arrayPlan.size(); i++)
					actionSequence.add(new Action(((JsonObject) arrayPlan.get(i)).get("name").toString()));
			} else {
				final Object resultObj = obj.get("result");
				errors += resultObj instanceof JsonObject ? " " + (String) ((JsonObject) resultObj).get("error") : resultObj.toString();
			}
		} catch (final DeserializationException e) {
			errors += " ParseException : " + e.getMessage();
		}
	}

}

File: Python version/specializations/solver_planning_domains/spd_plan.py
from languages.pddl.plan import Plan
from parsers.pddl.pddl_solvers_parser import PDDLSolversParser


class SPDPlan(Plan):
    """Represents a solution to a SPD problem."""

    def __init__(self, plan, error):
        super(SPDPlan, self).__init__(plan, error)

    def _parse(self):
        if self._errors or not self._output:
            return

        self._errors += PDDLSolversParser.parse_solver_planning_domains(
            self, self._output, True)
